recipes/libsolace/all/conanfile.py

```diff
@@ -1,66 +1,98 @@
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "5",
+            "apple-clang": "9",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
 
     def configure(self):
-        compiler_version = tools.Version(str(self.settings.compiler.version))
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
+    def validate(self):
         if self.settings.os == "Windows":
-          raise ConanInvalidConfiguration("This library is not yet compatible with Windows")
-        # Exclude compilers that claims to support C++17 but do not in practice
-        if (self.settings.compiler == "gcc" and compiler_version < "7") or \
-           (self.settings.compiler == "clang" and compiler_version < "5") or \
-           (self.settings.compiler == "apple-clang" and compiler_version < "9"):
-          raise ConanInvalidConfiguration("This library requires C++17 or higher support standard. {} {} is not supported".format(self.settings.compiler, self.settings.compiler.version))
-        if self.settings.compiler.cppstd and not self.settings.compiler.cppstd in self._supported_cppstd:
-          raise ConanInvalidConfiguration("This library requires c++17 standard or higher. {} required".format(self.settings.compiler.cppstd))
+            raise ConanInvalidConfiguration("This library is not yet compatible with Windows")
+
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["PKG_CONFIG"] = False
+        tc.cache_variables["SOLACE_GTEST_SUPPORT"] = False
+        tc.generate()
 
-    def _configure_cmake(self):
-        cmake = CMake(self, parallel=True)
-        cmake.definitions["PKG_CONFIG"] = "OFF"
-        cmake.definitions["SOLACE_GTEST_SUPPORT"] = "OFF"
-        cmake.configure(source_folder=self._source_subfolder)
-        return cmake
+    def _patch_sources(self):
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                        "include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)\nconan_basic_setup()",
+                        "")
```

Comment:
Interesting. It deserves an extra contribution to the upstream. I'll do it later.